libvast/src/systemd.cpp

```diff
@@ -0,0 +1,48 @@
+#include "vast/systemd.hpp"
+
+#include "vast/detail/posix.hpp"
+#include "vast/error.hpp"
+#include "vast/logger.hpp"
+
+#include <caf/detail/scope_guard.hpp>
+
+#include <cstdlib>
+#include <cstring>
+
+#include <sys/socket.h>
+
+namespace vast::systemd {
+
+#if VAST_LINUX
```

Comment:
Since this file is not conditionally compiled, does it make sense to rename it to something like `init_system.cpp` so we can provide implementations for launchd and rc.d later on if needed?

Reply:
I think it makes sense to have a more general header when we add support for launchd and rc.d, but I'm not sure we need to add it prematurely. We can still introduce it as soon as we actually have something to abstract away from.